.github/workflows/test.yml

```diff
@@ -28,11 +30,13 @@ jobs:
             puma: "5"
           - ruby: "2.7"
             puma: "4"
+    env:
+      PUMA_VERSION: ${{ matrix.puma }}
```

Comment:
This is necessary so that each run of the matrix picks up the correct version of puma. Otherwise they all were running against puma v6 even though the title suggested they were not.